app/models/salt_handler/orchestration_result.rb

```diff
@@ -19,11 +19,15 @@ def process_event
     when "orch.kubernetes", "orch.update"
       update_minions orchestration_succeeded: orchestration_succeeded
     when "orch.removal"
+      # Since this orchestration is parametrized is more correct to only update the minions that
+      # are specified in the orchestration parameters.
       if orchestration_succeeded
-        Minion.pending_removal.destroy_all
+        Minion.pending_removal.where(minion_id: (orchestration.params || {})["target"]).destroy_all
```

Comment:
as we currently don't have a way in velum to remove multiple minions at the same time, a more direct and explicit `find_by(minion_id: (orchestration.params || {})["target"]).destroy` would make it more readable

rest LGTM :+1:

Reply:
It was in purpose indeed, as this will work with a single target (as of now), and in the future if we change `target` to `targets` (array containing minions to remove) it will also work without problem with a minimum change.